edk2toollib/uefi/edk2/variable_policy.py

```diff
@@ -0,0 +1,171 @@
+# @file
+# Module contains helper classes and functions to work with Variable Policy structures
+# and substructures.
+#
+# Copyright (c) Microsoft Corporation
+#
+# SPDX-License-Identifier: BSD-2-Clause-Patent
+##
+
+import uuid
+import struct
+from edk2toollib.uefi.uefi_multi_phase import EfiVariableAttributes
+
+
+class VariableLockOnVarStatePolicy(object):
+    # typedef struct {
+    #     EFI_GUID Namespace;
+    #     UINT8    Value;
+    #     UINT8    Reserved;
+    #     // CHAR16   Name[];           // Variable Length Field
+    # } VARIABLE_LOCK_ON_VAR_STATE_POLICY;
+    _HdrStructFormat = "<16sBB"
+    _HdrStructSize = struct.calcsize(_HdrStructFormat)
+
+    def __init__(self):
+        self.Namespace = uuid.UUID(bytes=b'\x00' * 16)
+        self.Value = 0
+        self.Name = None
+
+    def __str__(self):
+        return "VARIABLE_LOCK_ON_VAR_STATE_POLICY(%s, %d, %s)" % (self.Namespace, self.Value, self.Name)
+
+    def decode(self, buffer):
+        """
+        load this object from a bytes buffer
+
+        return any remaining buffer
+        """
+        (_namespace, self.Value, _) = struct.unpack(
+            self._HdrStructFormat, buffer[:self._HdrStructSize])
+
+        self.Namespace = uuid.UUID(bytes_le=_namespace)
+
+        if len(buffer) > self._HdrStructSize:
+            self.Name = buffer[self._HdrStructSize:].decode(
+                'utf-16').strip('\x00')
+
+        return b''
```

Comment:
This is not what function header says it will do

Reply:
It is now, buddy boy.